ecssd_agent.go

```diff
@@ -371,13 +408,17 @@ func main() {
 	}
 	configuration.HostedZoneId = zoneId
 	metadataClient := ec2metadata.New(session.New())
-	hostname, err := metadataClient.GetMetadata("/hostname")
-	configuration.Hostname = hostname
+	localIP, err := metadataClient.GetMetadata("/local-ipv4")
+	configuration.Hostname = "ip-" + strings.Replace(localIP, ".", "-", 3) + "." + DNSName
```

Comment:
Right, but A records can also be weighted. So, if you set Weight to `0`, you can use the SetIdentifier.